### PR TITLE
[Modules] Updated Policy Assignments to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated Policy Assignments to new dependency approach

| Pipeline |
| - |
| [Authorization: PolicyAssignments](https://github.com/Azure/ResourceModules/actions/workflows/ms.authorization.policyassignments.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

**Comment by ahmadabdalla** [comment]
Pending change to the implementation approach:
MicrosoftTeams-image

**Comment by MrMCake** [comment]
> Pending change to the implementation approach: MicrosoftTeams-image

As per our discussion I'd suggest to push this back as neither the global.module.tests, nor especially the Set-ModuleReadMe function will be able to properly handle this alternative approach. A larger refactoring may be required.

File: modules/Microsoft.Authorization/policyAssignments/.test/mg/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,51 @@
+targetScope = 'managementGroup'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'apamgdef'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/policyAssignments/.test/mg/deploy.test.bicep:7]:
```suggestion
param serviceShort string = 'apamgcom'
```

File: modules/Microsoft.Authorization/policyAssignments/.test/rg.min/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,39 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(80)
+param resourceGroupName string = 'ms.xxx.xxx-${serviceShort}-rg'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/policyAssignments/.test/rg.min/deploy.test.bicep:8]:
```suggestion
param resourceGroupName string = 'ms.authorization.policyassignments-${serviceShort}-rg'
```

File: modules/Microsoft.Authorization/policyAssignments/.test/rg/dependencies.bicep [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+@description('Required. The name of the Managed Identity to create.')
+param managedIdentityName string
+
+resource managedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2018-11-30' = {
+    name: managedIdentityName
+    location: location
+}
+
+resource keyVault 'Microsoft.KeyVault/vaults@2022-07-01' = {
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/policyAssignments/.test/rg/dependencies.bicep:15]: We don't need a Key Vault for the dependencies

File: modules/Microsoft.Authorization/policyAssignments/.test/rg/deploy.test.bicep [hunk 4]
```diff
@@ -0,0 +1,78 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(80)
+param resourceGroupName string = 'ms.xxx.xxx-${serviceShort}-rg'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/policyAssignments/.test/rg/deploy.test.bicep:8]:
```suggestion
param resourceGroupName string = 'ms.authorization.policyassignments-${serviceShort}-rg'
```

File: modules/Microsoft.Authorization/policyAssignments/.test/rg/deploy.test.bicep [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'apargdef'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/policyAssignments/.test/rg/deploy.test.bicep:14]:
```suggestion
param serviceShort string = 'apargcom'
```

File: modules/Microsoft.Authorization/policyAssignments/.test/sub/deploy.test.bicep [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(80)
+param resourceGroupName string = 'ms.authorization.policyassignments-${serviceShort}-rg'
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'apasubdef'
```
Comment (ahmadabdalla) [inline modules/Microsoft.Authorization/policyAssignments/.test/sub/deploy.test.bicep:11]:
```suggestion
param serviceShort string = 'apasubcom'
```

**Review by ahmadabdalla — CHANGES_REQUESTED** [review]
Thanks Alex. Some minor suggestions.

File: modules/Microsoft.Authorization/policyAssignments/.test/rg/dependencies.bicep (hunk 3 above)
Comment (ahmadabdalla) [reply modules/Microsoft.Authorization/policyAssignments/.test/rg/dependencies.bicep:15]: sorry just noticed that we're using it for the not scopes section.

**Comment by ahmadabdalla** [comment]
@MrMCake please align the folder names to the proposed one from the role assignments module: `mg.common`

File: modules/Microsoft.Authorization/policyAssignments/.test/mg/deploy.test.bicep (hunk 1 above)
Comment (MrMCake) [reply modules/Microsoft.Authorization/policyAssignments/.test/mg/deploy.test.bicep]: done

File: modules/Microsoft.Authorization/policyAssignments/.test/rg.min/deploy.test.bicep (hunk 2 above)
Comment (MrMCake) [reply modules/Microsoft.Authorization/policyAssignments/.test/rg.min/deploy.test.bicep]: done

File: modules/Microsoft.Authorization/policyAssignments/.test/rg/deploy.test.bicep (hunk 5 above)
Comment (MrMCake) [reply modules/Microsoft.Authorization/policyAssignments/.test/rg/deploy.test.bicep]: done

File: modules/Microsoft.Authorization/policyAssignments/.test/sub/deploy.test.bicep (hunk 6 above)
Comment (MrMCake) [reply modules/Microsoft.Authorization/policyAssignments/.test/sub/deploy.test.bicep]: done

File: modules/Microsoft.Authorization/policyAssignments/.test/rg/deploy.test.bicep (hunk 4 above)
Comment (MrMCake) [reply modules/Microsoft.Authorization/policyAssignments/.test/rg/deploy.test.bicep]: done

**Review by ahmadabdalla — APPROVED** [review]

**Pull request merged**